### PR TITLE
snap: don't create empty Change with "Hold" state on disconnect

### PR DESCRIPTION
**Pull request by stolowski**
When using the abbreviated form of `snap disconnect snap:<plug|slot>` and there are no connections, we create an empty change with no tasks, and it remains in Hold state and produces the following error:
```
$ snap disconnect spotify:x11                                                 
error: change finished in status "Hold" with no error message
```
This PR fixes it. snap disconnect will not report any error if there are no connections for given plug or slot and an empty Change with Done status will be created. Note that in any other case (invalid plug or slot) we report an error.

### CODE REVIEW

File: daemon/api.go [hunk 1]
```diff
@@ -1864,6 +1864,11 @@ func changeInterfaces(c *Command, r *http.Request, user *auth.UserState) Respons
 	}
 
 	change := newChange(st, a.Action+"-snap", summary, tasksets, affected)
+	// Special case for disconnect: if no connections exist, we just report Done.
+	// Note that for every other case ResolveDisconnect will give an error.
+	if len(tasksets) == 0 {
```
Comment (zyga) [inline daemon/api.go]: As discussed IRL, I think this should behave exactly like "snap connect foo:bar froz:bot" when those things are already connected, that is do nothing (no tasks) successfully.

**Comment by chipaca** [comment]
Drop this in `response.go`
```go

// InterfacesUnchanged is an error responder used when an operation
// that would normally change interfaces finds it has nothing to do
func InterfacesUnchanged(format string, v ...interface{}) Response {
	res := &errorResult{
		Message: fmt.Sprintf(format, v...),
		Kind:    errorKindInterfacesUnchanged,
	}
	return &resp{
		Type:   ResponseTypeError,
		Result: res,
		Status: 400,
	}
}
```
(with an appropriate `kind` defined also there, and in `client/client.go`), and then return that

**Comment by chipaca** [comment]
You're missing the other half of the trick :-)
You're missing `client.ErrorKindInterfacesUnchanged` (in `client/client.go`), and
in `cmd/snap/cmd_disconnect.go`, where it says
```go
	id, err := cli.Disconnect(offer.Snap, offer.Name, use.Snap, use.Name)
	if err != nil {
		return erri
	}
```
check `err.(*client.Error).Kind == client.ErrorKindInterfacesUnchanged` (maybe have a client.IsUnchanged(error)` helper), in which case print a friendly non-error and exit.

**Comment by chipaca** [comment]
(oops closed in error)

**Comment by stolowski** [comment]
@Chipa Thanks, fixed.

**Pull request merged**